crates/gateway-client/src/lib.rs

```diff
@@ -446,6 +458,72 @@ impl ClientApi for Client {
     }
 }
 
+#[cfg(any(test, feature = "test-utils"))]
+pub mod utils {
+    use anyhow::Context;
+    use pathfinder_common::{BlockId, Chain};
+    use semver::Version;
+    use tokio::sync::OnceCell;
+
+    use crate::{Client, ClientApi};
+
+    /// Returns true if the latest block's version matches the requirement.
+    /// 
+    /// This is useful to add version checks for deprecation items or changes that
+    /// should be made when a certain version reaches a specifc network.
+    /// 
+    /// If the gateway query fails, then this still returns true - this is to prevent
+    /// failures due to rate limiting or if the gateway is down. In other words this
+    /// may give false positives if the gateway is unavailable for whatever reason.
+    pub async fn gateway_version_check(network: Chain, version_requirement: &str) -> bool {
```

Comment:
I am probably missing some context, but for me it is not intuitive what exactly the result of the check is (`bool`) and how it might be used. Propagating the error here seems like a reasonable option IMO.

Reply:
Because its exclusively meant to perform a version check in test code - i.e. this won't ever get used in production code. Essentially every usage would repeat the same steps which probably means it won't get used as its painful.